crates/dbsp/src/operator/chain_aggregate.rs

```diff
@@ -0,0 +1,49 @@
+use crate::{
+    dynamic::{DowncastTrait, DynData},
+    trace::BatchReaderFactories,
+    DBData, OrdIndexedZSet, RootCircuit, Stream, ZWeight,
+};
+
+impl<K, V> Stream<RootCircuit, OrdIndexedZSet<K, V>>
+where
+    K: DBData,
+    V: DBData,
+{
+    /// Aggregate whose value depends on the previous value of the aggregate
+    /// and changes to the input collection.
+    ///
+    /// Unlike general aggregates, such aggregates don't require storing the integral
+    /// of the input collection and hence require only O(1) memory per key.
+    ///
+    /// Examples include min and max over append-only collections.
+    ///
+    /// # Arguments
+    ///
+    /// * `finit` - returns the initial value of the aggregate.
+    /// * `fupdate` - updates the aggregate given its previous value and a
+    ///   new element of the group.
+    pub fn chain_aggregate<A, FInit, FUpdate>(
+        &self,
+        finit: FInit,
+        fupdate: FUpdate,
+    ) -> Stream<RootCircuit, OrdIndexedZSet<K, A>>
+    where
+        A: DBData,
+        FInit: Fn() -> A + 'static,
```

Comment:
can this work without an init?
there is no init for some data types

Reply:
I'll change the signature to `FInit: Fn(&V) -> A`, i.e., it gets initialized when it observes the first value..